cmds/core/gzip/gzip.go

```diff
@@ -32,15 +32,21 @@ func main() {
 		os.Exit(2)
 	}
 
-	var input []string
-	if opts.Stdin {
-		input = []string{"/dev/stdin"}
+	var input []gzip.File
+	args := cmdLine.Args()
+
+	if len(args) == 0 {
+		// no args given, compress stdin to stdout
+		opts.Stdin = true
+		opts.Stdout = true
```

Comment:
huh, I figured opts.Stdin and opts.Stdout should already be true here, because validate assigns them to be true if len(cmdLine.Args()) == 0, but I'm a bit confused by the check for ModeCharDevice there, too.

Reply:
I definitely agree with taking out the path checks, though, I'm just wondering if validate() has a bug

Reply:
I agree that setting Stdin and Stdout should be done in one place, so it's better to fix validate. Right now the check works for commands like this `echo "aaaaa" | ./gzip -f`, but it fails for `./gzip -f`, but i think that just checking for empty args in validate should be ok in this case?
```go
	if !moreArgs {
		o.Stdin = true
		// Since there's no filename to derive the output path from, only support
		// outputting to stdout when data is piped from stdin
		o.Stdout = true
	}
```